packages/maskbook/src/extension/background-script/EthereumServices/send.ts

```diff
@@ -43,13 +64,47 @@ export async function INTERNAL_send(
         account = currentAccountSettings.value,
         providerType = currentProviderSettings.value,
         rpc,
+        skipConfirmation = false,
     }: SendOverrides = {},
 ) {
     if (process.env.NODE_ENV === 'development' && debugModeSetting.value) {
         console.table(payload)
         console.debug(new Error().stack)
     }
 
+    // some rpc methods need to be confirmed by the user
+    if (
+        Flags.v2_enabled &&
+        !skipConfirmation &&
+        isRpcNeedToBeConfirmed(payload) &&
+        providerType === ProviderType.Maskbook
+    ) {
+        try {
+            await WalletRPC.pushUnconfirmedRequest(payload)
+        } catch (error: any) {
+            callback(error)
+            return
+        }
+
+        let target = ''
+
+        //TODO: Maybe there are other cases
+        switch (payload.method) {
+            case EthereumMethodType.ETH_SEND_TRANSACTION:
+                target = `popups.html#${DialogRoutes.GasSetting}`
+                break
+            case EthereumMethodType.PERSONAL_SIGN:
+                target = `popups.html#${DialogRoutes.WalletSignRequest}`
+                break
+            default:
+                break
+        }
+
+        if (target) {
+            window.open(browser.runtime.getURL(target), '', 'resizable,scrollbars,status,width=310,height=540')
```

Comment:
I'm worried about this. `window.open` requires [user activation](https://developers.google.com/web/updates/2019/01/user-activation) and the activation token might not share across different Web contexts. I guess it might not work on Firefox.

Reply:
It will be resolved by @nuanyang233 later.

Reply:
I tested it on firefox and I can't open the popups window with ` window.open`, Are there alternatives @Jack-Works ?